server/cache.js

```diff
@@ -57,46 +59,94 @@ Cache.prototype.start = function(){
  */
 
 Cache.prototype.poll = function *(){
-  debug('polling ecs...');
-  let ecs = this.ecs;
-
-  // retrieve the cluster definitions
-  let clusters = yield ecs.clusters();
-  clusters = clusters.clusters;
-  debug('received %d clusters', Object.keys(clusters).length);
-
-  // from all the clusters, retrieve the services
-  let serviceCalls = clusters.map(cluster => {
-    return ecs.services(cluster.clusterArn);
-  })
-  let services = yield Promise.all(serviceCalls);
-  services = flatten(services);
-  debug('received %d services', services.length);
-
-  // from all the tasks, get the versions running, then cache the result.
-  let taskCache = this.tasks;
-  let taskCalls = services.map(service => {
-    let task = taskCache.get(service.taskDefinition);
-    if (task) {
-      return Promise.resolve(task);
-    }
-    return ecs.task(service.taskDefinition).then(task => {
-      taskCache.set(service.taskDefinition, task);
-      return task;
+    debug('polling ecs...');
+    let ecs = this.ecs;
+    let ec2 = this.ec2;
+
+    // retrieve the cluster definitions
+    let clusters = yield ecs.clusters();
```

Comment:
I'm curious if we can split some of this stuff up into smaller calls? This method grew a bit, and I can see that if it had to be maintained with more functionaltiy in the future it will grow more.  Given we have a working version here I'm interested to see if we can take the blocks that are delineated with comments and make them smaller functions we can call.